proto/binlog/binlog.proto

```diff
@@ -10,21 +10,43 @@ option (gogoproto.unmarshaler_all) = true;
 
 // TableMutation contains mutations in a table.
 message TableMutation {
-    optional int64 table_id     = 1 [(gogoproto.nullable) = false];
-    repeated int64 inserted_ids = 2;
-    repeated int64 updated_ids  = 3;
-    repeated int64 deleted_ids  = 4;
+    optional int64 table_id      = 1 [(gogoproto.nullable) = false];
 
-    // For tables with non-interger types primary key.
-    repeated bytes deleted_pks = 5;
+    // For inserted rows and updated rows, we save all column values of the row.
+    repeated bytes inserted_rows = 2;
+    repeated bytes updated_rows  = 3;
 
-    // For tables without primary key.
-    repeated bytes deleted_rows = 6;
+    // If the table PK is handle, we can only save the id of the deleted row.
+    repeated int64 deleted_ids   = 4;
+
+    // If the table has PK but PK is not handle, we save the PK of the deleted row.
+    repeated bytes deleted_pks   = 5;
+
+    // If the table doesn't have PK, we save the row value of the deleted row.
+    repeated bytes deleted_rows  = 6;
+}
+
+enum BinlogType {
```

Comment:
Indeed, according to the design, when a transaction is committed, it should write two binlogs with different type.